amr-wind/equation_systems/vof/vof_advection.H

```diff
@@ -0,0 +1,77 @@
+
+#ifndef VOF_ADVECTION_H
+#define VOF_ADVECTION_H
+
+#include "amr-wind/equation_systems/vof/vof.H"
+#include "amr-wind/equation_systems/vof/SplitAdvection.H"
+
+namespace amr_wind {
+namespace pde {
+
+/** Godunov scheme for VOF
+ *  \ingroup icns
+ */
+
+template <>
+struct AdvectionOp<VOF, fvm::Godunov>
+{
+    AdvectionOp(PDEFields& fields_in, bool, bool)
+        : fields(fields_in)
+        , u_mac(fields_in.repo.get_field("u_mac"))
+        , v_mac(fields_in.repo.get_field("v_mac"))
+        , w_mac(fields_in.repo.get_field("w_mac"))
+    {}
+
+    void operator()(const FieldState, const amrex::Real dt)
+    {
+        static_assert(
+            VOF::ndim == 1, "Invalid number of components for scalar");
+        auto& repo = fields.repo;
+        auto& geom = repo.mesh().Geom();
+
+        auto& dof_field = fields.field;
+        auto bcrec_device = dof_field.bcrec_device();
+        //
+        // Advect volume using the Explicit Lagrangian onto-cell Sweeping method
+        // with PLIC reconstruction
+        //
+
+        // Define the sweep time
+        isweep += 1;
+        if (isweep > 3) isweep = 1;
```

Comment:
Why is this needed?

Reply:
Well it is not needed it can simply be isweep=isweep%3